tasks/configure-RedHat.yml

```diff
@@ -24,13 +24,16 @@
   stat:
     path: /etc/pki/tls/certs/localhost.crt
   register: localhost_cert
-  when: ansible_distribution_major_version | int >= 8
+  when:
+    - ansible_distribution_major_version | int >= 8
+    - not apache_remove_default_vhost_ssl
 
 - name: Ensure httpd certs are installed (RHEL 8 and later).
   command: /usr/libexec/httpd-ssl-gencerts
   when:
     - ansible_distribution_major_version | int >= 8
     - not localhost_cert.stat.exists
```

Comment:
This condition will fail because the previous task is not executed and thus the variable `localhost_cert` will not exist:

```
fatal: [localhost]: FAILED! => {"msg": "The conditional check 'not localhost_cert.stat.exists' failed. The error was: error while evaluating conditional (not localhost_cert.stat.exists): 'dict object' has no attribute 'stat'\n\nThe error appears to be in '/home/vagrant/ansible/roles/wcm_io_devops.apache/tasks/configure-RedHat.yml': line 31, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Ensure httpd certs are installed (RHEL 8 and later).\n  ^ here\n"}
```